addon/components/plaid-axis/component.js

```diff
@@ -30,50 +41,60 @@ export default Ember.Component.extend(GroupElement, {
    */
   scale: null,
 
+  /**
+   * The format used for the ticks for this axis.
+   * [See D3 docs for more details](https://github.com/d3/d3/wiki/SVG-Axes#tickFormat)
+   *
+   * @public
+   * @type {Function}
+   */
   tickFormat: null,
 
+  /**
+   * The inner tick size for the ticks for this axis.
+   * [See D3 docs for more details](https://github.com/d3/d3/wiki/SVG-Axes#innerTickSize)
+   *
+   * @public
+   * @type {Number}
+   */
+  tickSizeInner: 4,
+
+  /**
+   * The outer tick size for the ticks for this axis.
+   * [See D3 docs for more details](https://github.com/d3/d3/wiki/SVG-Axes#outerTickSize)
+   *
+   * @public
+   * @type {Number}
+   */
+  tickSizeOuter: 8,
+
   xOffset: 0,
 
   yOffset: 0,
 
-  didInsertElement() {
-    this._super();
-    this.groupElement = select(this.element);
-    scheduleOnce('afterRender', this, this.drawAxis);
-  },
-
   didReceiveAttrs() {
     scheduleOnce('afterRender', this, this.drawAxis);
   },
 
   drawAxis() {
-    let { y, x, xOffset, yOffset } = this.getProperties('y', 'x', 'xOffset', 'yOffset');
-    let scale = this.get('scale');
-    let orientation = this.get('orientation');
-    let tickFormat = this.get('tickFormat');
-    let ticks = this.get('ticks');
+    let { y, x, xOffset, yOffset, scale, orientation, tickFormat, ticks, tickSizeInner, tickSizeOuter } =
+      this.getProperties('y', 'x', 'xOffset', 'yOffset', 'scale', 'orientation', 'tickFormat', 'ticks', 'tickSizeInner', 'tickSizeOuter');
 
     let axis = this.createAxis(orientation, scale);
 
     axis.tickFormat(tickFormat);
-    axis.tickSizeOuter(8);
-    axis.tickSizeInner(4);
+    axis.tickSize(tickSizeInner, tickSizeOuter);
     axis.scale(scale);
 
     if (ticks) {
       axis.ticks(ticks);
     }
 
-    this.groupElement.call(axis);
-    this.groupElement.attr('transform', `translate(${x + xOffset}, ${y + yOffset})`);
+    this.selection.call(axis);
```

Comment:
It comes from the GroupElement mixin. It's the same change we made in
plaid-bar.
On Jun 21, 2016 7:38 AM, "Ivan Vanderbyl" <notifications@github.com> wrote:

> In addon/components/plaid-axis/component.js
> <https://github.com/ivanvanderbyl/maximum-plaid/pull/22#discussion_r67859363>
> :
>
> >      axis.scale(scale);
> >
> >      if (ticks) {
> >        axis.ticks(ticks);
> >      }
> >
> > -    this.groupElement.call(axis);
> > -    this.groupElement.attr('transform', `translate(${x + xOffset}, ${y + yOffset})`);
> > +    this.selection.call(axis);
>
> Where does this.selection get set?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ivanvanderbyl/maximum-plaid/pull/22/files/69d043cdf52444f5ef899e97cf4930b9cc73dbe3#r67859363>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AApQf7AerNjgrjjbshl-aRfRjphWTXzPks5qN9swgaJpZM4I5_vz>
> .
>